--- git status ---
On branch main
Your branch is up to date with 'origin/main'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/manipulation_env.py

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/manipulation_env.py b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/manipulation_env.py
index 6c343f3..2e07aa9 100644
--- a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/manipulation_env.py
+++ b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/manipulation_env.py
@@ -284,7 +284,7 @@ class ManipulationEnv(DirectRLEnv):
         ee_pose_error_mapped = ee_position_error_mapped #+ ee_orientation_error
 
         # end effector final velocity reward #TODO reduce even rot vel and acc?
-        should_freeze = ee_position_error < 0.0009 #ee position error in a radius of 3cm
+        should_freeze = ee_position_error < 0.0025 #ee position error in a radius of 5cm
         ee_final_velocity_error = torch.sum(torch.square(self._imu.data.lin_vel_b), dim=1)
         ee_final_velocity_error_mapped = torch.exp(-ee_final_velocity_error / 0.10)*should_freeze
 