--- git status ---
On branch main
Your branch is up to date with 'origin/main'.

Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   scripts/rsl_rl/__pycache__/cli_args.cpython-311.pyc
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/__pycache__/__init__.cpython-311.pyc
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/actuators/__pycache__/__init__.cpython-311.pyc
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/actuators/__pycache__/actuator_cfg.cpython-311.pyc
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/actuators/__pycache__/actuator_electric.cpython-311.pyc
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/actuators/__pycache__/actuator_hydraulic.cpython-311.pyc
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/assets/__pycache__/__init__.cpython-311.pyc
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/assets/__pycache__/aliengo_asset.cpython-311.pyc
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/assets/aliengo_asset.py
	deleted:    source/trash_collection_isaaclab/trash_collection_isaaclab/assets/aliengo_z1_nogripper_arm_fixed.usd
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/__pycache__/__init__.cpython-311.pyc
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/__pycache__/custom_curriculums.cpython-311.pyc
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/__pycache__/custom_events.cpython-311.pyc
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/__pycache__/supervised_learning_networks.cpython-311.pyc
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/__init__.cpython-311.pyc
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/aliengo_env_cfg.cpython-311.pyc
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/locomotion_env.cpython-311.pyc
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/agents/__pycache__/__init__.cpython-311.pyc
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/agents/__pycache__/rsl_rl_ppo_cfg.cpython-311.pyc
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/aliengo_env_cfg.py
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/arm_env_cfg.py
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/locomotion_env.py
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/manipulation_env.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	source/trash_collection_isaaclab/trash_collection_isaaclab/assets/aliengo_z1_nogripper_arm.usd
	source/trash_collection_isaaclab/trash_collection_isaaclab/assets/aliengo_z1_nogripper_arm_nocontact.usd

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/scripts/rsl_rl/__pycache__/cli_args.cpython-311.pyc b/scripts/rsl_rl/__pycache__/cli_args.cpython-311.pyc
index b097f26..6454214 100644
Binary files a/scripts/rsl_rl/__pycache__/cli_args.cpython-311.pyc and b/scripts/rsl_rl/__pycache__/cli_args.cpython-311.pyc differ
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/__pycache__/__init__.cpython-311.pyc b/source/trash_collection_isaaclab/trash_collection_isaaclab/__pycache__/__init__.cpython-311.pyc
index 1a6b058..6d6543d 100644
Binary files a/source/trash_collection_isaaclab/trash_collection_isaaclab/__pycache__/__init__.cpython-311.pyc and b/source/trash_collection_isaaclab/trash_collection_isaaclab/__pycache__/__init__.cpython-311.pyc differ
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/actuators/__pycache__/__init__.cpython-311.pyc b/source/trash_collection_isaaclab/trash_collection_isaaclab/actuators/__pycache__/__init__.cpython-311.pyc
index ebfb2e5..b3b68fb 100644
Binary files a/source/trash_collection_isaaclab/trash_collection_isaaclab/actuators/__pycache__/__init__.cpython-311.pyc and b/source/trash_collection_isaaclab/trash_collection_isaaclab/actuators/__pycache__/__init__.cpython-311.pyc differ
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/actuators/__pycache__/actuator_cfg.cpython-311.pyc b/source/trash_collection_isaaclab/trash_collection_isaaclab/actuators/__pycache__/actuator_cfg.cpython-311.pyc
index f4514ae..e95afa4 100644
Binary files a/source/trash_collection_isaaclab/trash_collection_isaaclab/actuators/__pycache__/actuator_cfg.cpython-311.pyc and b/source/trash_collection_isaaclab/trash_collection_isaaclab/actuators/__pycache__/actuator_cfg.cpython-311.pyc differ
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/actuators/__pycache__/actuator_electric.cpython-311.pyc b/source/trash_collection_isaaclab/trash_collection_isaaclab/actuators/__pycache__/actuator_electric.cpython-311.pyc
index ec0f676..9a907a0 100644
Binary files a/source/trash_collection_isaaclab/trash_collection_isaaclab/actuators/__pycache__/actuator_electric.cpython-311.pyc and b/source/trash_collection_isaaclab/trash_collection_isaaclab/actuators/__pycache__/actuator_electric.cpython-311.pyc differ
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/actuators/__pycache__/actuator_hydraulic.cpython-311.pyc b/source/trash_collection_isaaclab/trash_collection_isaaclab/actuators/__pycache__/actuator_hydraulic.cpython-311.pyc
index d195670..4667a2a 100644
Binary files a/source/trash_collection_isaaclab/trash_collection_isaaclab/actuators/__pycache__/actuator_hydraulic.cpython-311.pyc and b/source/trash_collection_isaaclab/trash_collection_isaaclab/actuators/__pycache__/actuator_hydraulic.cpython-311.pyc differ
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/assets/__pycache__/__init__.cpython-311.pyc b/source/trash_collection_isaaclab/trash_collection_isaaclab/assets/__pycache__/__init__.cpython-311.pyc
index 700cd84..2b5471b 100644
Binary files a/source/trash_collection_isaaclab/trash_collection_isaaclab/assets/__pycache__/__init__.cpython-311.pyc and b/source/trash_collection_isaaclab/trash_collection_isaaclab/assets/__pycache__/__init__.cpython-311.pyc differ
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/assets/__pycache__/aliengo_asset.cpython-311.pyc b/source/trash_collection_isaaclab/trash_collection_isaaclab/assets/__pycache__/aliengo_asset.cpython-311.pyc
index 42fd7c8..17b90c8 100644
Binary files a/source/trash_collection_isaaclab/trash_collection_isaaclab/assets/__pycache__/aliengo_asset.cpython-311.pyc and b/source/trash_collection_isaaclab/trash_collection_isaaclab/assets/__pycache__/aliengo_asset.cpython-311.pyc differ
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/assets/aliengo_asset.py b/source/trash_collection_isaaclab/trash_collection_isaaclab/assets/aliengo_asset.py
index 966c6c6..b3e7eb9 100644
--- a/source/trash_collection_isaaclab/trash_collection_isaaclab/assets/aliengo_asset.py
+++ b/source/trash_collection_isaaclab/trash_collection_isaaclab/assets/aliengo_asset.py
@@ -72,7 +72,7 @@ ALIENGO_ARM_ACTUATOR_CFG = IdentifiedActuatorElectricCfg(
 ALIENGO_CFG = ArticulationCfg(
     prim_path=None,
     spawn=sim_utils.UsdFileCfg(
-        usd_path=f"{ISAAC_ASSET_DIR}/aliengo_z1_nogripper_arm_fixed.usd",
+        usd_path=f"{ISAAC_ASSET_DIR}/aliengo_z1_nogripper_arm_nocontact.usd",
         activate_contact_sensors=True,
         rigid_props=sim_utils.RigidBodyPropertiesCfg(
             disable_gravity=False,
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/assets/aliengo_z1_nogripper_arm_fixed.usd b/source/trash_collection_isaaclab/trash_collection_isaaclab/assets/aliengo_z1_nogripper_arm_fixed.usd
deleted file mode 100644
index 49cdf27..0000000
Binary files a/source/trash_collection_isaaclab/trash_collection_isaaclab/assets/aliengo_z1_nogripper_arm_fixed.usd and /dev/null differ
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/__pycache__/__init__.cpython-311.pyc b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/__pycache__/__init__.cpython-311.pyc
index 1abd621..d0946b7 100644
Binary files a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/__pycache__/__init__.cpython-311.pyc and b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/__pycache__/__init__.cpython-311.pyc differ
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/__pycache__/custom_curriculums.cpython-311.pyc b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/__pycache__/custom_curriculums.cpython-311.pyc
index 14f4158..c8a75c6 100644
Binary files a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/__pycache__/custom_curriculums.cpython-311.pyc and b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/__pycache__/custom_curriculums.cpython-311.pyc differ
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/__pycache__/custom_events.cpython-311.pyc b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/__pycache__/custom_events.cpython-311.pyc
index 54e40e7..1faac01 100644
Binary files a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/__pycache__/custom_events.cpython-311.pyc and b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/__pycache__/custom_events.cpython-311.pyc differ
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/__pycache__/supervised_learning_networks.cpython-311.pyc b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/__pycache__/supervised_learning_networks.cpython-311.pyc
index 7aa7280..40759c0 100644
Binary files a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/__pycache__/supervised_learning_networks.cpython-311.pyc and b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/__pycache__/supervised_learning_networks.cpython-311.pyc differ
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/__init__.cpython-311.pyc b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/__init__.cpython-311.pyc
index c7440a7..6e43314 100644
Binary files a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/__init__.cpython-311.pyc and b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/__init__.cpython-311.pyc differ
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/aliengo_env_cfg.cpython-311.pyc b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/aliengo_env_cfg.cpython-311.pyc
index 952a8ad..1df2d4d 100644
Binary files a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/aliengo_env_cfg.cpython-311.pyc and b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/aliengo_env_cfg.cpython-311.pyc differ
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/locomotion_env.cpython-311.pyc b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/locomotion_env.cpython-311.pyc
index e3b6a20..50ece46 100644
Binary files a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/locomotion_env.cpython-311.pyc and b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/locomotion_env.cpython-311.pyc differ
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/agents/__pycache__/__init__.cpython-311.pyc b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/agents/__pycache__/__init__.cpython-311.pyc
index 4783bf9..3bfa2f3 100644
Binary files a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/agents/__pycache__/__init__.cpython-311.pyc and b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/agents/__pycache__/__init__.cpython-311.pyc differ
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/agents/__pycache__/rsl_rl_ppo_cfg.cpython-311.pyc b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/agents/__pycache__/rsl_rl_ppo_cfg.cpython-311.pyc
index 3b825dd..1fa2926 100644
Binary files a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/agents/__pycache__/rsl_rl_ppo_cfg.cpython-311.pyc and b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/agents/__pycache__/rsl_rl_ppo_cfg.cpython-311.pyc differ
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/aliengo_env_cfg.py b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/aliengo_env_cfg.py
index c6cf0bc..04d2c2c 100644
--- a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/aliengo_env_cfg.py
+++ b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/aliengo_env_cfg.py
@@ -134,7 +134,6 @@ class AliengoFlatEnvCfg(DirectRLEnvCfg):
     action_scale = 0.5
     action_space = 12
     observation_space = 48
-    observation_space += 12 #12 are the arm joints pos and vel
     observation_space += 2 # 2 are for pose commands
     state_space = 0
 
@@ -149,6 +148,8 @@ class AliengoFlatEnvCfg(DirectRLEnvCfg):
         single_observation_space = observation_space # Placeholder. Later we may add map, but only from the latest obs
         observation_space *= history_length
 
+    observation_space += 6 # we add arm joints info now
+
     use_imu = False
 
     use_cuncurrent_state_est = False
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/arm_env_cfg.py b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/arm_env_cfg.py
index 3fc5b4a..d254301 100644
--- a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/arm_env_cfg.py
+++ b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/arm_env_cfg.py
@@ -209,19 +209,24 @@ class ArmFlatEnvCfg(DirectRLEnvCfg):
         mesh_prim_paths=["/World/ground"],
     )
 
-    # an imu sensor in case we don't want any state estimator
-    imu = ImuCfg(prim_path="/World/envs/env_.*/Robot/base", debug_vis=True)
-
+    # HACK to have gripper position
+    imu = ImuCfg(
+        prim_path="/World/envs/env_.*/Robot/link05", 
+        offset=ImuCfg.OffsetCfg(
+            pos=(0.18, 0, 0)
+        ), 
+        debug_vis=False)
+    
     # some marker for visualization
     goal_object_cfg = VisualizationMarkersCfg(
         prim_path="/Visuals/myMarkers",
         markers={
             "sphere": sim_utils.SphereCfg(
-                radius=0.1,
+                radius=0.03,
                 visual_material=sim_utils.PreviewSurfaceCfg(diffuse_color=(0.0, 1.0, 0.0)),
             ),
         },
-)
+    )
 
     # scene
     scene: InteractiveSceneCfg = InteractiveSceneCfg(num_envs=4096, env_spacing=4.0, replicate_physics=True)
@@ -258,6 +263,7 @@ class ArmFlatEnvCfg(DirectRLEnvCfg):
     
     # Tracking reward scale
     ee_pose_reward_scale = 1.5
+    ee_final_velocity_reward_scale = 1.5
     
     # Joint reward scale
     joints_torque_reward_scale = -2.5e-6
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/locomotion_env.py b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/locomotion_env.py
index cd04357..0a07dc2 100644
--- a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/locomotion_env.py
+++ b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/locomotion_env.py
@@ -43,8 +43,10 @@ class LocomotionEnv(DirectRLEnv):
 
         # X/Y linear velocity and yaw angular velocity commands
         self._velocity_commands = torch.zeros(self.num_envs, 3, device=self.device)
-        self._pose_commands = torch.zeros(self.num_envs, 2, device=self.device) # roll pitch. #TODO height
+        self._pose_commands = torch.zeros(self.num_envs, 2, device=self.device) # pitch height
 
+        # Arm fixed position
+        self._joints_arm_fixed_pos = torch.zeros(self.num_envs, 6, device=self.device)
 
         # Swing peak
         self._swing_peak = torch.tensor([0.0, 0.0, 0.0, 0.0], device=self.device).repeat(self.num_envs,1)
@@ -127,6 +129,7 @@ class LocomotionEnv(DirectRLEnv):
         self._hip_ids_robot, _ = self._robot.find_bodies(".*hip")
         self._ids_joints_order = self._robot.find_joints(name_keys=self.cfg.desired_joints_order, preserve_order=True)[0]
         self._ids_only_legs_joints_order = self._robot.find_joints(name_keys=self.cfg.desired_joints_order[0:12], preserve_order=True)[0]
+        self._ids_only_arms_joints_order = self._robot.find_joints(name_keys=self.cfg.desired_joints_order[12:18], preserve_order=True)[0]
 
 
     def _setup_scene(self):
@@ -176,8 +179,8 @@ class LocomotionEnv(DirectRLEnv):
 
 
     def _apply_action(self):
-        #TODO randomize pd target arms?
         processed_actions_with_arm = torch.zeros(self.num_envs, 18, device=self.device)
+        processed_actions_with_arm[:, self._ids_only_arms_joints_order] = self._joints_arm_fixed_pos.clone()
         processed_actions_with_arm[:, self._ids_only_legs_joints_order] = self._processed_actions
         self._robot.set_joint_position_target(processed_actions_with_arm)
 
@@ -225,8 +228,8 @@ class LocomotionEnv(DirectRLEnv):
                     projected_gravity_b,
                     self._velocity_commands,
                     self._pose_commands,
-                    self._robot.data.joint_pos[:,self._ids_joints_order] - self._robot.data.default_joint_pos[:,self._ids_joints_order],
-                    self._robot.data.joint_vel[:,self._ids_joints_order],
+                    self._robot.data.joint_pos[:,self._ids_only_legs_joints_order] - self._robot.data.default_joint_pos[:,self._ids_only_legs_joints_order],
+                    self._robot.data.joint_vel[:,self._ids_only_legs_joints_order],
                     self._actions,
                     clock_data,
                 )
@@ -238,6 +241,10 @@ class LocomotionEnv(DirectRLEnv):
             #the bottom element is the newest observation!!
             self._observation_history = torch.cat((self._observation_history[:,1:,:], obs.unsqueeze(1)), dim=1)
             obs = torch.flatten(self._observation_history, start_dim=1)
+        
+        # Add joint arm info
+        joints_arm = self._robot.data.joint_pos[:,self._ids_only_arms_joints_order] - self._robot.data.default_joint_pos[:,self._ids_only_arms_joints_order]
+        obs = torch.cat((obs, joints_arm), dim=-1)
 
 
         # Add heightmap data to obs if needed
@@ -267,9 +274,6 @@ class LocomotionEnv(DirectRLEnv):
             observations["critic"] = torch.cat((obs, obs_critic), dim=-1)
         # ------------------------------------------------------------------------------------------
 
-
-
-
         return observations
 
 
@@ -281,7 +285,7 @@ class LocomotionEnv(DirectRLEnv):
         height_data_scanner = torch.clip(height_data_scanner, min=-5, max=5) # Handle inf values
         mean_height_ray = torch.mean(height_data_scanner, dim=1)
 
-        height_error = torch.square(self.cfg.desired_base_height + mean_height_ray - self._robot.data.root_state_w[:, 2])
+        height_error = torch.square(self.cfg.desired_base_height + mean_height_ray + self._pose_commands[:,1] - self._robot.data.root_state_w[:, 2])
         height_error_mapped = torch.exp(-height_error / 0.01)
 
 
@@ -323,7 +327,7 @@ class LocomotionEnv(DirectRLEnv):
         root_pitch_w = torch.atan2(torch.sin(root_pitch_w), torch.cos(root_pitch_w))
         
         #base_orientation =  torch.square(terrain_pitch - root_pitch_w)# + torch.square(0 - root_roll_w)
-        base_orientation =  torch.square(terrain_pitch + self._pose_commands[:,0] - root_pitch_w)# + torch.square(0 + self._pose_commands[:,1] - root_roll_w)
+        base_orientation =  torch.square(terrain_pitch + self._pose_commands[:,0] - root_pitch_w) + torch.square(0 + root_roll_w)
 
 
         # angular velocity x/y tracking
@@ -556,7 +560,8 @@ class LocomotionEnv(DirectRLEnv):
         self._velocity_commands[env_ids, 0] *= 0.5
         self._velocity_commands[env_ids, 1] *= 0.25 
         self._velocity_commands[env_ids, 2] *= 0.3 
-        self._pose_commands[env_ids] = torch.zeros_like(self._pose_commands[env_ids])
+        self._pose_commands[env_ids, 0] = torch.zeros_like(self._pose_commands[env_ids,0]).uniform_(-0.3, 0.3)
+        self._pose_commands[env_ids, 1] = torch.zeros_like(self._pose_commands[env_ids,1]).uniform_(-0.2, 0.0)
 
         # Reset swing peak
         self._swing_peak[env_ids] = torch.tensor([0.0, 0.0, 0.0, 0.0], device=self.device)
@@ -576,7 +581,17 @@ class LocomotionEnv(DirectRLEnv):
 
         # Reset robot state
         joint_pos = self._robot.data.default_joint_pos[env_ids]
+        joint_pos[:, self._ids_only_arms_joints_order] += torch.zeros_like(joint_pos[:, self._ids_only_arms_joints_order]).uniform_(-3.14, 3.14)
+        # we need to project them inside the robots limits!
+        joints_limits = self._robot.data.default_joint_pos_limits
+        joints_arm_limits = joints_limits[:,self._ids_only_arms_joints_order]
+        joint_pos[:, self._ids_only_arms_joints_order] = torch.clamp(joint_pos[:, self._ids_only_arms_joints_order], joints_arm_limits[0,:,0], joints_arm_limits[0,:,1])
+        
+        # we save the arm positions to keep them fixed during the episode via PD control
+        self._joints_arm_fixed_pos[env_ids] = joint_pos[:, self._ids_only_arms_joints_order].clone()
+        
         joint_vel = self._robot.data.default_joint_vel[env_ids]
+        
         default_root_state = self._robot.data.default_root_state[env_ids]
         default_root_state[:, :3] += self._terrain.env_origins[env_ids]
         default_root_state[:, 3:7] = math_utils.random_yaw_orientation(env_ids.shape[0], device=self.device)
@@ -613,8 +628,11 @@ class LocomotionEnv(DirectRLEnv):
 
         # Stop and small pose commands
         rest_time = self.episode_length_buf >= self.max_episode_length - 150
-        self._velocity_commands[:, :3] *= ~rest_time.unsqueeze(1).expand(-1, 3)       
-        self._pose_commands = torch.zeros_like(self._pose_commands).uniform_(-0.3, 0.3) * rest_time.unsqueeze(1).expand(-1, 2)     
+        specific_rest_time = self.episode_length_buf == self.max_episode_length - 100
+        self._velocity_commands[:, :3] *= ~rest_time.unsqueeze(1).expand(-1, 3)
+        self._pose_commands[:, 0] = self._pose_commands[:, 0] * ~specific_rest_time + torch.zeros_like(self._pose_commands[:,0]).uniform_(-0.3, 0.3) * specific_rest_time
+        self._pose_commands[:, 1] = self._pose_commands[:, 1] * ~specific_rest_time + torch.zeros_like(self._pose_commands[:,1]).uniform_(-0.2, 0.0) * specific_rest_time
+        
 
         # Took some envs, and put to zero the vel
         if self.num_envs > 100:
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/manipulation_env.py b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/manipulation_env.py
index 8e13250..d1a2f79 100644
--- a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/manipulation_env.py
+++ b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/manipulation_env.py
@@ -90,6 +90,7 @@ class ManipulationEnv(DirectRLEnv):
             key: torch.zeros(self.num_envs, dtype=torch.float, device=self.device)
             for key in [
                 "track_ee_exp",
+                "track_ee_final_vel_exp",
 
                 "undesired_contacts",
                 "action_rate_l2",
@@ -271,11 +272,19 @@ class ManipulationEnv(DirectRLEnv):
         ROT_W2H = math_utils.matrix_from_quat(math_utils.yaw_quat(self._initial_root_quat))
         ee_position_commands_local_w = torch.matmul(ROT_W2H, self._ee_commands[:, :3].unsqueeze(2))
         ee_position_commands_w = ee_position_commands_local_w[:,:,0] + self._robot.data.default_root_state[:,0:3] + self.scene.env_origins
-        next_ee_position_w = self._robot.data.body_pos_w[:, self._ee_id_robot, :3] + self._robot.data.body_lin_vel_w[:, self._ee_id_robot, :] * self.step_dt
+        next_ee_position_w = self._imu.data.pos_w[:,:3].reshape((self._imu.data.pos_w.shape[0],1,3)) + self._robot.data.body_lin_vel_w[:, self._ee_id_robot, :] * self.step_dt
+        #next_ee_position_w = self._robot.data.body_pos_w[:, self._ee_id_robot, :3] + self._robot.data.body_lin_vel_w[:, self._ee_id_robot, :] * self.step_dt
         #ee_position_error = torch.sum(torch.square(ee_position_commands_w - (self._robot.data.body_pos_w[:, self._ee_id_robot, :3]).reshape((self._robot.data.body_pos_w.shape[0],3))), dim=1)
         ee_position_error = torch.sum(torch.square(ee_position_commands_w - next_ee_position_w.reshape((self._robot.data.body_pos_w.shape[0],3))), dim=1)
         ee_position_error_mapped = torch.exp(-ee_position_error / 0.10)
-        ee_pose_error_mapped = ee_position_error_mapped #+ ee_orientation_error_mapped
+
+        #ee_orientation_error = torch.sum(self._robot.data.body_quat_w[:, self._ee_id_robot] - self._ee_commands[:, 3:7], dim=1)
+        ee_pose_error_mapped = ee_position_error_mapped #+ ee_orientation_error
+
+        # end effector final velocity reward #TODO reduce even rot vel and acc?
+        should_freeze = ee_position_error < 0.0009 #ee position error in a radius of 3cm
+        ee_final_velocity_error = torch.sum(torch.square(self._imu.data.lin_vel_b), dim=1)
+        ee_final_velocity_error_mapped = torch.exp(-ee_final_velocity_error / 0.10)*should_freeze
 
         # action rate
         action_rate = torch.sum(torch.square(self._actions - self._previous_actions), dim=1)
@@ -344,6 +353,7 @@ class ManipulationEnv(DirectRLEnv):
 
         rewards = {
             "track_ee_exp": ee_pose_error_mapped * self.cfg.ee_pose_reward_scale * self.step_dt,
+            "track_ee_final_vel_exp": ee_final_velocity_error_mapped * self.cfg.ee_final_velocity_reward_scale * self.step_dt,
 
             "undesired_contacts": contacts * self.cfg.undersired_contact_reward_scale * self.step_dt,
             "action_rate_l2": action_rate * self.cfg.action_rate_reward_scale * self.step_dt,
@@ -460,6 +470,7 @@ class ManipulationEnv(DirectRLEnv):
         ee_position_commands_local_w = torch.matmul(ROT_W2H, self._ee_commands[:, :3].unsqueeze(2))
         goal_pos = ee_position_commands_local_w[:,:,0] + self._robot.data.default_root_state[:,0:3] + self.scene.env_origins
         self.goal_markers.visualize(goal_pos)
+        #self.goal_markers.visualize(self._imu.data.pos_w[:,:3])
 
 
     def _get_locomotion_policy_action(self, pose_commands):