--- git status ---
On branch main
Your branch is up to date with 'origin/main'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/aliengo_env_cfg.cpython-311.pyc
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/locomotion_env.cpython-311.pyc
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/aliengo_env_cfg.py
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/arm_env_cfg.py
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/locomotion_env.py
	modified:   source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/manipulation_env.py

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/aliengo_env_cfg.cpython-311.pyc b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/aliengo_env_cfg.cpython-311.pyc
index 40dd3e4..f0e69bd 100644
Binary files a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/aliengo_env_cfg.cpython-311.pyc and b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/aliengo_env_cfg.cpython-311.pyc differ
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/locomotion_env.cpython-311.pyc b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/locomotion_env.cpython-311.pyc
index 38ac487..131626d 100644
Binary files a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/locomotion_env.cpython-311.pyc and b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/__pycache__/locomotion_env.cpython-311.pyc differ
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/aliengo_env_cfg.py b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/aliengo_env_cfg.py
index 46224f6..c6cf0bc 100644
--- a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/aliengo_env_cfg.py
+++ b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/aliengo_env_cfg.py
@@ -273,13 +273,15 @@ class AliengoFlatEnvCfg(DirectRLEnvCfg):
                            'FL_calf_joint', 'FR_calf_joint', 'RL_calf_joint', 'RR_calf_joint',
                            'arm_joint1', 'arm_joint2', 'arm_joint3', 'arm_joint4', 'arm_joint5', 'arm_joint6']
 
-    # Desired gait
-    desired_gait = "trot" #crawl, pace, multigait
-
     # Desired tracking variables
     desired_base_height = 0.35
     desired_feet_height = 0.05
 
+    # Desired step freq and duty factor (if periodic gait is used)
+    desired_step_freq = 1.4
+    desired_duty_factor = 0.65
+    desired_phase_offset = [0.0, 0.5, 0.5, 0.0] #FL, FR, RL, RR
+
     # Desired clip actions
     desired_clip_actions = 3.0
     
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/arm_env_cfg.py b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/arm_env_cfg.py
index 69350d6..662f990 100644
--- a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/arm_env_cfg.py
+++ b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/arm_env_cfg.py
@@ -108,17 +108,13 @@ class ArmFlatEnvCfg(DirectRLEnvCfg):
     episode_length_s = 20.0
     decimation = 4
     action_scale = 0.5
-    action_space = 6
+    action_space = 6 + 2 # 6 for the arm, 2 for robot pose commands
     observation_space = 18 + 18 # 12 are the arm joints pos and vel
     observation_space += 9 # 9 the pose linear vel and angular, and orientation 
     observation_space += 3 # ee goal
     observation_space += 6 # action
     state_space = 0
 
-    use_clock_signal = False
-    if(use_clock_signal):
-        observation_space += 4
-
 
     # observation history
     use_observation_history = True
@@ -128,18 +124,6 @@ class ArmFlatEnvCfg(DirectRLEnvCfg):
         observation_space *= history_length
 
 
-    use_imu = False
-
-    use_cuncurrent_state_est = False
-    if(use_cuncurrent_state_est):
-        cuncurrent_state_est_output_space = 3 #lin_vel_b
-        single_cuncurrent_state_est_observation_space = single_observation_space
-        cuncurrent_state_est_observation_space = observation_space
-        cuncurrent_state_est_batch_size = 32
-        cuncurrent_state_est_train_epochs = 500
-        cuncurrent_state_est_lr = 1e-3
-        cuncurrent_state_est_ep_saving_interval = 1000
-
     use_rma = False
     if(use_rma):
         rma_output_space = 12 # P gain
@@ -251,9 +235,6 @@ class ArmFlatEnvCfg(DirectRLEnvCfg):
                            'arm_joint1', 'arm_joint2', 'arm_joint3', 'arm_joint4', 'arm_joint5', 'arm_joint6']
 
 
-    # Desired gait
-    desired_gait = "trot" #crawl, pace, multigait
- 
     # Desired clip actions
     desired_clip_actions = 3.0
     
@@ -273,6 +254,52 @@ class ArmFlatEnvCfg(DirectRLEnvCfg):
     action_smoothness_reward_scale = -0.001
 
 
+    # Loading the locomotion policy --------------------------------------------------------------------------------
+    try:
+        import sys
+        import os 
+        dir_path = os.path.dirname(os.path.realpath(__file__))
+        locomotion_policy_folder_path = dir_path + "/../../../../../../tested_policies/aliengo_with_z1/policy_folder_path"
+        cuncurrent_state_est_network_path = locomotion_policy_folder_path + "/exported/cuncurrent_state_estimator.pth"
+        rma_network_path = locomotion_policy_folder_path + "/exported/rma.pth"
+
+        # Load specific training parameters
+        import yaml 
+        with open(locomotion_policy_folder_path + "/params/env.yaml", "r") as file:
+            locomotion_policy_env_cfg = yaml.unsafe_load(file)
+
+        # state estimator locomotion
+        use_imu = locomotion_policy_env_cfg["use_imu"]
+
+        use_cuncurrent_state_est = locomotion_policy_env_cfg["use_cuncurrent_state_est"]
+        if(use_cuncurrent_state_est):
+            cuncurrent_state_est_output_space = locomotion_policy_env_cfg["cuncurrent_state_est_output_space"]
+            single_cuncurrent_state_est_observation_space = locomotion_policy_env_cfg["single_observation_space"]
+            cuncurrent_state_est_observation_space = locomotion_policy_env_cfg["observation_space"]
+        
+        # action locomotion
+        use_filter_actions_locomotion = locomotion_policy_env_cfg["use_filter_actions"]
+        desired_clip_actions_locomotion = locomotion_policy_env_cfg["desired_clip_actions"]
+        action_scale_locomotion = locomotion_policy_env_cfg["action_scale"]
+        single_action_space_locomotion = locomotion_policy_env_cfg["single_action_space"]
+
+        # periodic gait locomotion
+        use_clock_signal = locomotion_policy_env_cfg["use_clock_signal"]
+        # Desired step freq and duty factor (if periodic gait is used)
+        desired_step_freq = locomotion_policy_env_cfg["desired_step_freq"]
+        desired_duty_factor = locomotion_policy_env_cfg["desired_duty_factor"]
+        desired_phase_offset = locomotion_policy_env_cfg["desired_phase_offset"]
+
+        # observation locomotion
+        observation_space_locomotion = locomotion_policy_env_cfg["observation_space"]
+        use_observation_history_locomotion = locomotion_policy_env_cfg["use_observation_history"]
+        history_length_locomotion = locomotion_policy_env_cfg["history_length"]
+        if(use_observation_history_locomotion):
+            single_observation_space_locomotion = locomotion_policy_env_cfg["single_observation_space"]
+    except:
+        print("Error loading the locomotion policy parameters")  
+    # -----------------------------------------------------------------------------------------------------------------
+
 
 
 import isaaclab.terrains as terrain_gen
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/locomotion_env.py b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/locomotion_env.py
index 9f5a25c..cd04357 100644
--- a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/locomotion_env.py
+++ b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/locomotion_env.py
@@ -53,24 +53,9 @@ class LocomotionEnv(DirectRLEnv):
         self._desired_hip_offset = torch.tensor([-self.cfg.desired_hip_offset, self.cfg.desired_hip_offset, -self.cfg.desired_hip_offset, self.cfg.desired_hip_offset], device=self.device)
         
         # Periodic gait
-        if(cfg.desired_gait == "trot"):
-            self._step_freq = 1.4
-            self._duty_factor = 0.65
-            self._phase_offset = torch.tensor([0.0, 0.5, 0.5, 0.0], device=self.device).repeat(self.num_envs,1)
-            self._velocity_gait_multiplier = 1.0
-        elif(cfg.desired_gait == "crawl"):
-            self._step_freq = 0.5
-            self._duty_factor = 0.8
-            self._phase_offset = torch.tensor([0.0, 0.5, 0.75, 0.25], device=self.device).repeat(self.num_envs,1)
-            self._velocity_gait_multiplier = 0.5
-        elif(cfg.desired_gait == "pace"):
-            self._step_freq = 1.4
-            self._duty_factor = 0.7
-            self._phase_offset = torch.tensor([0.8, 0.3, 0.8, 0.3], device=self.device).repeat(self.num_envs,1)
-            self._velocity_gait_multiplier = 1.0
-        elif(cfg.desired_gait == "multigait"):
-            #TODO: implement multigait
-            raise NotImplementedError("Multigait not implemented yet")
+        self._step_freq = torch.tensor(self.cfg.desired_step_freq, device=self.device)
+        self._duty_factor = torch.tensor(self.cfg.desired_duty_factor, device=self.device)
+        self._phase_offset = torch.tensor(self.cfg.desired_phase_offset, device=self.device).repeat(self.num_envs,1)
         self._phase_signal = self._phase_offset.clone()# + self.step_dt * self._step_freq * torch.rand(self.num_envs, 1, device=self.device)*10.
         self._phase_signal = self._phase_signal % 1.0
 
@@ -568,7 +553,7 @@ class LocomotionEnv(DirectRLEnv):
         
         # Sample new commands
         self._velocity_commands[env_ids] = torch.zeros_like(self._velocity_commands[env_ids]).uniform_(-1.0, 1.0)
-        self._velocity_commands[env_ids, 0] *= 0.5 * self._velocity_gait_multiplier
+        self._velocity_commands[env_ids, 0] *= 0.5
         self._velocity_commands[env_ids, 1] *= 0.25 
         self._velocity_commands[env_ids, 2] *= 0.3 
         self._pose_commands[env_ids] = torch.zeros_like(self._pose_commands[env_ids])
@@ -621,7 +606,7 @@ class LocomotionEnv(DirectRLEnv):
     def _get_new_random_commands(self):
         resample_time = self.episode_length_buf == self.max_episode_length - 300
         commands_resample = torch.zeros_like(self._velocity_commands).uniform_(-1.0, 1.0)
-        commands_resample[:, 0] *= 0.5 * self._velocity_gait_multiplier
+        commands_resample[:, 0] *= 0.5
         commands_resample[:, 1] *= 0.25 
         commands_resample[:, 2] *= 0.3 
         self._velocity_commands[:, :3] = self._velocity_commands[:, :3] * ~resample_time.unsqueeze(1).expand(-1, 3) + commands_resample * resample_time.unsqueeze(1).expand(-1, 3)
diff --git a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/manipulation_env.py b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/manipulation_env.py
index 50475c3..ccf30d6 100644
--- a/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/manipulation_env.py
+++ b/source/trash_collection_isaaclab/trash_collection_isaaclab/tasks/locomotion/manipulation_env.py
@@ -7,6 +7,7 @@ from __future__ import annotations
 
 import gymnasium as gym
 import torch
+import onnxruntime as ort
 
 import isaaclab.envs.mdp as mdp
 import isaaclab.sim as sim_utils
@@ -49,27 +50,19 @@ class ManipulationEnv(DirectRLEnv):
 
 
         # Periodic gait
-        if(cfg.desired_gait == "trot"):
-            self._step_freq = 1.4
-            self._duty_factor = 0.65
-            self._phase_offset = torch.tensor([0.0, 0.5, 0.5, 0.0], device=self.device).repeat(self.num_envs,1)
-            self._velocity_gait_multiplier = 1.0
-        elif(cfg.desired_gait == "crawl"):
-            self._step_freq = 0.5
-            self._duty_factor = 0.8
-            self._phase_offset = torch.tensor([0.0, 0.5, 0.75, 0.25], device=self.device).repeat(self.num_envs,1)
-            self._velocity_gait_multiplier = 0.5
-        elif(cfg.desired_gait == "pace"):
-            self._step_freq = 1.4
-            self._duty_factor = 0.7
-            self._phase_offset = torch.tensor([0.8, 0.3, 0.8, 0.3], device=self.device).repeat(self.num_envs,1)
-            self._velocity_gait_multiplier = 1.0
-        elif(cfg.desired_gait == "multigait"):
-            #TODO: implement multigait
-            raise NotImplementedError("Multigait not implemented yet")
+        self._step_freq = torch.tensor(self.cfg.desired_step_freq, device=self.device)
+        self._duty_factor = torch.tensor(self.cfg.desired_duty_factor, device=self.device)
+        self._phase_offset = torch.tensor(self.cfg.desired_phase_offset, device=self.device).repeat(self.num_envs,1)
         self._phase_signal = self._phase_offset.clone()# + self.step_dt * self._step_freq * torch.rand(self.num_envs, 1, device=self.device)*10.
         self._phase_signal = self._phase_signal % 1.0
 
+        # Observation history locomotion
+        self._observation_history_locomotion = torch.zeros(self.num_envs, cfg.locomotion_policy_env_cfg["history_length"], cfg.locomotion_policy_env_cfg["single_observation_space"], device=self.device)
+
+        self._locomotion_policy = ort.InferenceSession(cfg.locomotion_policy_env_cfg["policy_path"] + "/exported/policy.onnx")
+        self._previous_actions_locomotion = torch.zeros(
+            self.num_envs, gym.spaces.flatdim(self.single_action_space_locomotion), device=self.device
+        )
 
         # Logging
         self._episode_sums = {
@@ -151,11 +144,13 @@ class ManipulationEnv(DirectRLEnv):
 
 
     def _apply_action(self):
-        processed_actions_with_arm = torch.zeros(self.num_envs, 18, device=self.device)
-        processed_actions_with_arm[:, self._ids_only_arms_joints_order] = self._processed_actions
-
+        arm_commands = self._processed_actions[:, 0:6]
+        pose_commands = self._processed_actions[:, 6:8]
         # Get locomotion policy action
-        locomotion_actions = self._get_locomotion_policy_action()
+        locomotion_actions = self._get_locomotion_policy_action(pose_commands)
+        
+        processed_actions_with_arm = torch.zeros(self.num_envs, 18, device=self.device)
+        processed_actions_with_arm[:, self._ids_only_arms_joints_order] = arm_commands
         processed_actions_with_arm[:, self._ids_only_legs_joints_order] = locomotion_actions
         self._robot.set_joint_position_target(processed_actions_with_arm)
 
@@ -363,11 +358,75 @@ class ManipulationEnv(DirectRLEnv):
         self._ee_commands[:, :3] = self._ee_commands[:, :3] * ~resample_time_2.unsqueeze(1).expand(-1, 3) + commands_resample * resample_time_2.unsqueeze(1).expand(-1, 3)
 
 
-    def _get_locomotion_policy_action(self,):
-        return torch.zeros(self.num_envs, 12, device=self.device)
+    def _get_locomotion_policy_action(self, pose_commands):
+        # Observation --------------------------------------------------------------------------------------
+        clock_data = None
+        if(self.cfg.use_clock_signal):
+            clock_data = torch.vstack([self._phase_signal[:,0], self._phase_signal[:,1], self._phase_signal[:,2], self._phase_signal[:,3]]).T
+            clock_data[:, :] = -1.0
+
+        # Choosing the main source of observation
+        if(self.cfg.use_cuncurrent_state_est):
+            # If Cuncurrent SE/Learned State Estimator, we predict linear and angular vel from IMU
+            velocity_b = self._get_cuncurrent_state_estimation(clock_data)
+            angular_velocity_b = self._imu.data.ang_vel_b
+            projected_gravity_b = self._imu.data.projected_gravity_b
+        elif(self.cfg.use_imu):
+            # Using directly the IMU
+            velocity_b = self._imu.data.lin_acc_b
+            angular_velocity_b = self._imu.data.ang_vel_b
+            projected_gravity_b = self._imu.data.projected_gravity_b
+        else:
+            #Using a model-based state estimation
+            velocity_b = self._robot.data.root_lin_vel_b
+            angular_velocity_b = self._robot.data.root_ang_vel_b
+            projected_gravity_b = self._robot.data.projected_gravity_b
+
+        velocity_commands = torch.zeros(self.num_envs, 3, device=self.device)
+        
+        
+        # Standard Obs for the Actor/Critic
+        obs = torch.cat(
+            [
+                tensor
+                for tensor in (
+                    velocity_b,
+                    angular_velocity_b,
+                    projected_gravity_b,
+                    velocity_commands,
+                    pose_commands,
+                    self._robot.data.joint_pos[:,self._ids_joints_order] - self._robot.data.default_joint_pos[:,self._ids_joints_order],
+                    self._robot.data.joint_vel[:,self._ids_joints_order],
+                    self._actions,
+                    clock_data,
+                )
+                if tensor is not None
+            ],
+            dim=-1,
+        )
+        if(self.cfg.use_observation_history_locomotion):
+            #the bottom element is the newest observation!!
+            self._observation_history_locomotion = torch.cat((self._observation_history_locomotion[:,1:,:], obs.unsqueeze(1)), dim=1)
+            obs = torch.flatten(self._observation_history_locomotion, start_dim=1)
+
+
+        self._actions_locomotion = self._locomotion_policy.run(None, {'obs': obs})[0][0]
+
+        # Clip the action
+        self._actions_locomotion = torch.clamp(self._actions_locomotion, -self.cfg.desired_clip_actions_locomotion, self.cfg.desired_clip_actions_locomotion)
+
+        # Filter the action
+        if(self.use_filter_actions_locomotion):
+            alpha = 0.8
+            temp = alpha * self._actions_locomotion + (1 - alpha) * self._previous_actions_locomotion
+            self._processed_actions_locomotion = self.cfg.action_scale_locomotion * temp + self._robot.data.default_joint_pos[:,self._ids_only_arms_joints_order]
+        else:
+            self._processed_actions_locomotion = self.cfg.action_scale_locomotion * self._actions_locomotion + self._robot.data.default_joint_pos[:,self._ids_only_arms_joints_order]
+
+        return self._processed_actions_locomotion
 
 
-    def _get_concurrent_state_estimation(self, clock_data):
+    def _get_cuncurrent_state_estimation(self, clock_data):
         return torch.zeros(self.num_envs, 3, device=self.device)
 
 